File: ChainOfResponsibility/C++/ErrorLogger.cpp
#include "ErrorLogger.hpp"
#include <iostream>

ErrorLogger::ErrorLogger(Logger* nextLoggerProcessor) : Logger(nextLoggerProcessor)
{
}

void ErrorLogger::logMessage(int logLevel, std::string message)
{
    if (logLevel == Logger::ERROR)
    {
        std::cout << "ERROR: " << message << std::endl;
    }
    else
    {
        Logger::logMessage(logLevel, message);
    }
}

File: ChainOfResponsibility/Java/ErrorLogger.java
public class ErrorLogger extends Logger {
    ErrorLogger(Logger nextLogProcessor) {
        super(nextLogProcessor);
    }

    @Override
    public void log(int logLevel, String message) {
        if (logLevel == ERROR) {
            System.out.println("ERROR: " + message);
        }
        if (nextLogProcessor != null) {
            nextLogProcessor.log(logLevel, message);
        }
    }
}